repro/BUILD

```diff
@@ -21,4 +21,10 @@ deploy_jar(
         ":repro",
     ],
     main = "repro.Main",
+    exclude_files = [
+        r"^META-INF/.*\.SF$",
```

Comment:
The `META-INF/SIGNINGC.SF` in the deploy jar when `//:google-cloud-pubsub` dependencies are added ended up being this one:
```
Signature-Version: 1.0
SHA-256-Digest-Manifest-Main-Attributes: ASqUKgrRbC3DfKBNwMfHFOcDvF08h
 TJ1XA0TQ0CmOhw=
SHA-256-Digest-Manifest: QKAgAUe/0Eaz3gLp4jk0I46YK0PUfbfhxu8IivVuc9o=
Created-By: 1.8.0_131 (Oracle Corporation)

Name: org/conscrypt/ConscryptCertStore.class
SHA-256-Digest: 56F0dVReWxs63y8MOCAHR94hVt2O4WeUwbbBA6o2+FY=
...
```

I'm not sure if Pants v2 just kept 1 of many same-name files here from dependency jars that composed the deploy jar or if there was only one.

Reply:
So, basically, java meant to say "I'm ignoring jar.jar because it has a bad signature". And thats why the obviously present `repro/Main.class` in that jar was not found.

Reply:
It looks like you can fix modern Pants here: https://github.com/pantsbuild/pants/blob/2a72e0f27e5ac72d88a93c7375fb6b03d81b6351/src/python/pants/jvm/target_types.py#L819-L853

Reply:
Fixed in https://github.com/pantsbuild/pants/pull/20943, FWIW